File: app/src/main/java/com/example/pge/repositories/loginApi.java
package com.example.pge.repositories;

public class loginApi {
    val email: String,
    val password: String
}

data class LoginResponse(
        val success: Boolean,
        val token: String?,
        val user: User?
)

data class User(
        val id: Int,
        val name: String
)

File: app/src/main/java/com/example/pge/data/network/LoginApi.kt
package com.example.pge.data.network

import com.example.pge.models.LoginRequest
import com.example.pge.models.LoginResponse
import com.example.pge.models.UserResponse
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.POST
import retrofit2.http.GET

interface LoginApi {

    @POST("login")
    suspend fun login(@Body request: LoginRequest): Response<LoginResponse>

    @GET("auth/me")
    suspend fun getUser(): Response<UserResponse>

}
